src/utils/WithToggleControl.jsx

```diff
@@ -0,0 +1,49 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+/**
+ * Provides boolean isChecked prop to wrapped component.
+ * Prop name can also be customized if need be
+ *
+ * @param {React.element} ToggleComponent - the component to wrap
+ * @returns {React.element}
+ */
+export default function withToggleControl(WrappedComponent, propName = 'isChecked') {
+	/**
+	 * @module WithToggle
+	 */
+	class WithToggle extends React.Component {
+		constructor(props) {
+			super(props);
+
+			this.state = { isActive: props['isActive'] };
+			this.onToggleBool = this.toggleBool.bind(this);
+		}
+		toggleBool() {
+			this.setState({ isActive: !this.state.isActive });
+		}
+		render() {
+
+			const newProps = {
+				...this.props,
+				isActive: this.state.isActive,
+				toggleActive: this.onToggleBool
+			};
```

Comment:
very minor, but I think in general it would be preferable to avoid creating a new object here and instead pass the two additional props to the `<WrappedComponent>` directly

```jsx
<WrappedComponent {...this.props} isActive={this.state.isActive} toggleActive={this.onToggleBool} />
```